Addition of free(output) has hosed this for me

I am not enough of an expert in memory management issues in C to suggest why, but merging #89 not only turned up [failures across the board](https://github.com/kspalaiologos/bzip3/actions/runs/4362077805) in CI (which could have been GitHub Actions having a bad moment) but it also fails for me locally when I build from Git HEAD:

``` console
$ make roundtrip
rm -f ./LICENSE2
./bzip3 -v -feb 6 ./LICENSE ./LICENSE.bz3
make: *** [Makefile:1206: roundtrip] Segmentation fault (core dumped)
```

It looks like the specific CI failures vary some by platform, but universally it either fails to build or fails to pass a round trip check (when it gets that far).

--- a/src/main.c
+++ b/src/main.c
@@ -690,7 +690,6 @@ int main(int argc, char * argv[]) {
 
     output_des = mode != MODE_TEST ? open_output(output, force) : NULL;
     input_des = open_input(input);
-    free(output);
 
     int r = process(input_des, output_des, mode, block_size, workers, verbose, input);
 

